File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[][]}
 */

var levelOrder = function(root) {
    
    if(!root) return [];
    
    let result = [];
    let deque = [root];
    
    while(deque.length > 0){
        const length = deque.length;
        let level = [];
        
        for(let i = 0; i < length; i++){
            const current = deque.shift();
            level.push(current.val);
            
            if(current.left) deque.push(current.left);
            if(current.right) deque.push(current.right);
        }
        result.push(level)
    }
    return result;
};

File: 0102-binary-tree-level-order-traversal/0102-binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        if root == None:
            return []
        
        result = []
        que = collections.deque([root])
        
        while que:
            que_len = len(que)
            level = []
            
            for i in range(que_len):
                current = que.popleft()
                level.append(current.val)
                
                if current.left:
                    que.append(current.left)
                    
                if current.right:
                    que.append(current.right)
                
            result.append(level)
            
        return result